Mnemonic prefixes break pycodestyle --diff execution

In the `diff` section in my gitconfig I have an option `mnemonicprefix` set to `true`. This means that git will use `i/w/c/o` instead of `a/b/c/d` in the generated diff. Apparently that breaks execution of `pycodestyle` when `--diff` argument is provided.

Command:

    git diff -U0 --staged | pycodestyle --diff --max-line-length=120

Output:

    ./i/path.py:1:1: E902 IOError: [Errno 2] No such file or directory: './i/path.py'

To fix it I could use `--no-prefix` in git but I am wondering whether this should be handled by `pycodestyle` itself.

--- a/pycodestyle.py
+++ b/pycodestyle.py
@@ -1538,7 +1538,9 @@ def parse_udiff(diff, patterns=None, parent='.'):
             rv[path].update(range(row, row + nrows))
         elif line[:3] == '+++':
             path = line[4:].split('\t', 1)[0]
-            if path[:2] == 'b/':
+            # Git diff will use (i)ndex, (w)ork tree, (c)ommit and (o)bject
+            # instead of a/b/c/d as prefixes for patches
+            if path[:2] in ('b/', 'w/', 'i/'):
                 path = path[2:]
             rv[path] = set()
     return dict([(os.path.join(parent, path), rows)

--- a/testsuite/test_shell.py
+++ b/testsuite/test_shell.py
@@ -191,3 +191,13 @@ class ShellTestCase(unittest.TestCase):
         self.assertFalse(errcode)
         self.assertFalse(stdout)
         self.assertFalse(stderr)
+
+        for index, diff_line in enumerate(diff_lines, 0):
+            diff_line = diff_line.replace('a/', 'i/')
+            diff_lines[index] = diff_line.replace('b/', 'w/')
+
+        self.stdin = '\n'.join(diff_lines)
+        stdout, stderr, errcode = self.pycodestyle('--diff')
+        self.assertFalse(errcode)
+        self.assertFalse(stdout)
+        self.assertFalse(stderr)

